libraries/aws_guardduty_detectors.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'aws_backend'
+
+class AwsGuardDutyDetectors < AwsResourceBase
+  name 'aws_guardduty_detectors'
+  desc 'Verifies settings for a Detector in bulk'
+  example '
+    describe aws_guardduty_detectors do
+      it { should exist }
+    end
+  '
+
+  attr_reader :table, :detector_rows, :api_response
+
+  FilterTable.create
+             .register_column(:detector_ids, field: :detector_ids)
+             .install_filter_methods_on_resource(self, :table)
+
+  def initialize(opts = {})
+    super(opts)
+    validate_parameters
+    @table = fetch_data
+  end
+
+  def fetch_data
+    detector_rows = []
+    pagination_options = {}
+    loop do
+      catch_aws_errors do
+        @api_response = @aws.guardduty_client.list_detectors(pagination_options)
+      end
+      return [] if !api_response || api_response.empty?
```

Comment:
This should return `[]` when `api_response.detector_ids.empty?` as well . 
Otherwise `fetch_data` returns `[{:detector_ids=>[]}]` when there is no detector, and, makes the resource respond `true`, due to not empty `@table`, to `exist` matcher.

Reply:
Updated in latest commit